query/src/main/java/org/infinispan/query/affinity/AffinityIndexManager.java

```diff
@@ -155,26 +179,23 @@ private void handleOwnershipAcquired() {
 
    @Override
    public void flushAndReleaseResources() {
-      Address lockHolder = getLockHolder(getIndexName());
-      if (lockHolder != null) {
-         InfinispanDirectoryProvider directoryProvider = (InfinispanDirectoryProvider) getDirectoryProvider();
-         int activeDeleteTasks = directoryProvider.getActiveDeleteTasks();
-         boolean wasInterrupted = false;
-         while (activeDeleteTasks > 0) {
-            try {
-               Thread.sleep(POLL_WAIT);
-               logDebug("Waiting for pending delete tasks, remaining: %s", activeDeleteTasks);
-               activeDeleteTasks = directoryProvider.getActiveDeleteTasks();
-            } catch (InterruptedException e) {
-               wasInterrupted = true;
-            }
-         }
-         if (wasInterrupted) {
-            Thread.currentThread().interrupt();
+      InfinispanDirectoryProvider directoryProvider = (InfinispanDirectoryProvider) getDirectoryProvider();
+      int activeDeleteTasks = directoryProvider.pendingDeleteTasks();
+      boolean wasInterrupted = false;
+      while (activeDeleteTasks > 0) {
+         try {
+            Thread.sleep(POLL_WAIT);
```

Comment:
What will happen if there will be a deadlocked delete task? If I read the code correctly we will be spinning here forever? Is there any way to avoid this?